playground/src/routes.js

```diff
@@ -7,15 +7,32 @@ import {
   permissions,
 } from '@commercetools-frontend/permissions';
 import StateMachinesList from './components/state-machines-list';
+import StateMachinesDetails from './components/state-machines-details';
 import reducers from './reducers';
 
 // FIXME: import it from AppShell
 const PageUnauthorized = () => <div>{'Unauthorized'}</div>;
 PageUnauthorized.displayName = 'PageUnauthorized';
 
-const ApplicationRoutes = () => (
+const ApplicationRoutes = props => (
   <InjectReducers id="state-machines" reducers={reducers}>
     <Switch>
+      <Route
+        path={`${props.match.path}/:id`}
+        render={routerProps => (
+          <RestrictedByPermissions
+            permissions={[permissions.ViewStates, permissions.ManageStates]}
+            unauthorizedComponent={PageUnauthorized}
+            shouldMatchSomePermissions={true}
+          >
+            <StateMachinesDetails
+              id={routerProps.match.params.id}
+              projectKey={routerProps.match.params.projectKey}
+              backToListPath={props.match.url}
```

Comment:
We often started to pass down functions for this kind of navigation in the MC. So here `goToStateMachinesList`. It ultimatively eases things a bit cause a single parent is in control of routing.

Reply:
Yes but in this case we pass the string to the `LinkButton` component `to` prop, which is a string.

Reply:
Ok.